.github/workflows/release.yml

```diff
@@ -38,30 +41,70 @@ jobs:
     runs-on: "ubuntu-22.04"
     steps:
       - uses: actions/checkout@v3
+
       - name: Setup build env
         run: ./scripts/setup-linux.sh
-      - uses: actions-rust-lang/setup-rust-toolchain@v1
+      
+      - name: Setup rust toolchain
+        uses: actions-rust-lang/setup-rust-toolchain@v1
         env:
           RUST_CACHE_KEY_OS: rust-release-cache-${{ needs.parse.outputs.crate }}-${{ matrix.arch }}
         with:
           rustflags: '' #Disable.  By default this action sets environment variable is set to -D warnings.  We manage this in the Makefile
+      
       - name: Setup cross-rs
         run: ./scripts/setup-cross.sh ${{ matrix.arch }}-unknown-linux-musl
+      
       - name: Setup build profile
         shell: bash
         run: echo "OPT_PROFILE=release" >> ${GITHUB_ENV}
+      
+      - name: Setup cosign for signing
+        uses: sigstore/cosign-installer@v3.3.0
+        with:
+          cosign-release: 'v2.2.2'
+      
       - name: Build
         timeout-minutes: 20
         run: make build-${{ needs.parse.outputs.runtime }}
+
       - name: Test
         if: ${{ matrix.arch == 'x86_64' }}
         timeout-minutes: 10
         run: make test-${{ needs.parse.outputs.runtime }}
+
+      - name: Sign the binary
+        if: ${{ needs.parse.outputs.runtime != 'wasm' }}
+        run: |
+          make dist-${{ needs.parse.outputs.runtime }}
+          # Check if there's any files to archive as tar fails otherwise
+          if stat dist/bin/* >/dev/null 2>&1; then
```

Comment:
What's the scenario where there are no files after running make `dist-*`

Reply:
You meant afer `make build-*`? 

If we release `containerd-shim-wasm-test-modules`, I think there will be no files in `disk/bin`.